nix/overlay.nix

```diff
@@ -0,0 +1,64 @@
+# Override some packages and utilities in 'pkgs'
+# and make them available globally via callPackage.
+#
+# For more details see:
+# - https://nixos.wiki/wiki/Overlays
+# - https://nixos.org/nixos/nix-pills/callpackage-design-pattern.html
+
+final: prev: let
+  inherit (prev) config stdenv callPackage recurseIntoAttrs makeOverridable fetchurl lib writeShellScriptBin __splicedPackages;
+in rec {
+  linuxdeployqt = callPackage ./pkgs/linuxdeployqt/default.nix { };
+
+  # Copyied from d9424d2191d6439a276b69ae1fd0a800586135ca
+  # 2018-07-27 -> 2020-12-31
+  # TODO: override and upgrade
+  # Copy is uses because of initial complexity of package override (probably due to fuse override)
+  appimagekit = callPackage ./pkgs/appimagekit/default.nix { };
+
+  # Requirement from Makefile - 3.19
+  cmake_3_19 = prev.cmake.overrideAttrs ( attrs : rec {
+    version = "3.19.7";
+
+    src = fetchurl {
+      url = "${attrs.meta.homepage}files/v${lib.versions.majorMinor version}/cmake-${version}.tar.gz";
+      # compare with https://cmake.org/files/v${lib.versions.majorMinor version}/cmake-${version}-SHA-256.txt
+      sha256 = "sha256-WKFfDVagr8zDzFNxI0/Oc/zGyPnb13XYmOUQuDF1WI4=";
+    };
+  });
+
+  # Copyied from bootstrap121 from 020300a756e75ea9ce86a8ab5ee259c31e28ed43
+  # - autoPatchelfHook is disabled
+  # - development/compilers/go/print-hashes.sh 1.21.11
+  # TODO: compile, not binary
+  # Binary is used because of initial complexity of both package override and copy from newer nixpkgs
+  go_1_21 = callPackage ./pkgs/go/bootstrap121.nix { };
+
+  # Fix for linuxdeployqt running ldd from nix with system shell
+  # ERROR: findDependencyInfo: "/bin/sh: /nix/store/HASH-glibc-2.31-74/lib/libc.so.6: version `GLIBC_2.33' not found (required by /bin/sh)\n/bin/sh: /nix/store/0c7c96gikmzv87i7lv3vq5s1cmfjd6zf-glibc-2.31-74/lib/libc.so.6: version `GLIBC_2.34' not found (required by /bin/sh)"
+  # $ head $(which ldd)
+  # #! /bin/sh
+  lddWrapped = writeShellScriptBin "ldd" ''
+    "${final.bash}/bin/sh" "${final.glibc.bin}/bin/ldd" "$@"
+  '';
+
+  # Qt 5.15.8 copy from 76973ae3b30a88ea415f27ff53809ab8f452e2ec
+  # Edited:
+  # - temporary break Darwin support
+  # - remove unsupported testers, env., config.allowAliases
+  # - mkDerivation without finalAttrs
+  # - change fetch* parameter from hash to sha256, rmove fetchLFS
+  # - fix makeSetupHook
+  # - switch from makeScopeWithSplicing back to makeScope
+  # See diff for a full list of changes
```

Comment:
I worry that as we keep adding changes to this copy of `qt` derivation we'll end up in a situation where we can't use the `qt` available in `nixpkgs` due to these differences. It's good that you commented on the changes here, but it would be difficult to compare the changes. Could we maybe add to this comment from which commit of `nixpkgs` we copied this `qt` derivation so we can do the comparison more easily?

Reply:
Currently it these 2 commits:
```
❯ git log nix/pkgs/qt-5/
commit 77c282f51cd0cd286c0a0a5eb00b1be8855eaeea
Author: Anton Iakimov <yakimant@gmail.com>
Date:   Wed Apr 10 13:50:08 2024 +0200

    ci: fix qtwebengine issue

commit b4075c62d2afadec5523a6a9634510badd65e7c1
Author: Anton Iakimov <yakimant@gmail.com>
Date:   Fri Jan 19 16:23:26 2024 +0100

    ci: ammend qt derivation to run on 20.09

commit d17d77ae066c46284e7e0d90ed31213e330e3eb4
Author: Anton Iakimov <yakimant@gmail.com>
Date:   Fri Jan 19 13:30:40 2024 +0100

    ci: copy qt 5.15.8 package
```

Thcecking the diffs you can see, what exactly was changed.